modules/openapi-generator/src/main/resources/typescript/api/api.mustache

```diff
@@ -151,11 +157,11 @@ export class {{classname}}RequestFactory extends BaseAPIRequestFactory {
         {{/bodyParam}}
 
         {{#hasAuthMethods}}
-        let authMethod = null;
+        let authMethod: SecurityAuthentication | undefined;
         {{/hasAuthMethods}}
         {{#authMethods}}
         // Apply auth methods
-        authMethod = _config.authMethods["{{name}}"]
+        authMethod = this.defaultSecurityAuthentication || _config.authMethods["{{name}}"]
```

Comment:
Doesn't this make it impossible to overwrite the `defaultSecurityAuthentication` through the `_options` object in the method call?